### PR TITLE
(feat) recovery run

### PR DESCRIPTION
**Pull request by Eli-BB8**
Recovery run - compare data with kartoffel data

### CODE REVIEW

File: mocks/mockGen.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ let esUsers = [];
 // Generating mi and tz lists
 for (let i = 0; i < 300; i++) {
     tzs.push(utils.generateID());
-    mis.push(faker.random.number({'min': 100000,'max': 999999999}));
+    mis.push(faker.random.number({'min': 100000,'max': 999999999}).toString());
 }
 
 fs.writeFileSync("./lists/miList.json", JSON.stringify(mis))
```
Comment (weizmm) [inline mocks/mockGen.js]: delete

File: mocks/lists/tzList.json [hunk 2]
```diff
@@ -1 +1 @@
...
..."803618537","471734806","121873343","434083614","837881192","248358830","760733444","929708030","275290294","221083140","508154275","522914266","174730622","383761491","474374337","488661612","339337628","278938931","346994353","444364590","464609460","988924973","173594326","530507169","968167825","922572771","428739585","961065109","324365089","166710814","449853670","155502412","161329834","104099429","582224085","252518113","440423630","584037097","159168319","451863047","446541781","190210633","305499899","502638273","365134048","750054470","133725127","831015540","596446161","791630411","117868216","322831660","163859689","409522646","423517218","676679558","696446483","630678514","603663568","623838166","555708726","781825732","335939229","485884845","610421489","816679492","584925291","175474584","753958180","100515154","654556455","589617836","635229784","885942094","509526919","871877122","793945551","820674281","999789027","703588558","215866922","414538363","295892442","833768468","883230591","311467377","394003313","852151299","885176701","439124348","218037455","774038509","367270048","371032087","166495515","458659828","201321304","651080665","975142183","187004825"]
```
Comment (weizmm) [inline mocks/lists/tzList.json]: delete

File: mocks/lists/miList.json [hunk 3]
```diff
@@ -1 +1 @@
...
...950515","379904485","441860914","808383957","121378103","837431763","190662315","507108763","605670411","578995063","791735586","523990424","997954862","577571735","377473676","814341313","598064703","965308530","956081477","476707383","880584342","707477550","672695511","856780741","368465470","384605392","59648606","375958776","885536997","202185999","127120290","86213930","350694897","374707211","971986407","790284444","101946883","222595459","849447553","184116580","468771414","355441351","749179405","353242678","581942888","701587673","858002794","942975797","420103512","728839512","500449056","211571269","954937286","26633172","944070108","452235786","131363589","10564377","527206482","725686111","911582059","354185058","796057986","391155062","387551262","211919003","630313375","10024356","97842121","61580557","85794636","788944246","342463237","437062714","292496966","951659602","364731496","688901595","870595490","610535716","468874058","266048072","623052916","374245532","905609349","772244594","125282590","783121454","373460331","851384133","587183340","430460098","628792221","689478998","145853533","264112403","608788866","191869607","161851942","665027362","784210183"]
```
Comment (weizmm) [inline mocks/lists/miList.json]: delete

File: mocks/mockGen.js [hunk 4]
```diff
@@ -24,40 +24,42 @@ fs.writeFileSync("./lists/tzList.json", JSON.stringify(tzs))
...
-    "hr": utils.randomElement(dataTypes.UNIT)
+    hr: utils.randomElement(dataTypes.UNIT)
     })
     telephones.push({
-        "mi": mi[i],
-        "telephone": utils.generateNumberBody(),
-        "ktelephone": utils.generateNumberPrefix(),
-        "telephoneType": faker.random.number({'min': 1,'max': 2})
-    })
+        mi: mi[i],
```
Comment (weizmm) [inline mocks/mockGen.js]: take PN from mis

File: mocks/mockGen.js [hunk 5]
```diff
@@ -76,21 +78,28 @@ for (let i = 0; i < 100; i++) {
...
+// Generating es employee/unemployee objects
 for (let i = 0; i < 50; i++) {
     let user = {};
-    user.stype = utils.randomElement(dataTypes.SERVICE_TYPE);
-    user.firstName = faker.name.firstName();
-    user.lastName = faker.name.lastName();
     user.tz = utils.randomElement([utils.generateID(), tzs[250 + i]]);
 
+    // employee
     if (user.tz === tzs[250 + i]) {
```
Comment (weizmm) [inline mocks/mockGen.js]: change to parameters

File: src/util/data_synchronizeData.js [hunk 6]
```diff
@@ -1,29 +1,14 @@
-const p = require('../config/paths');
 const fn = require('../config/fieldNames');
 const axios = require("axios");
 const saveAsFile = require('../util/saveAsFile');
 const dataComparison = require('../util/dataComparison');
-const akaDataManipulate = require('./akaDataManipulate');
-const logger = require('./logger');
+const getRawData = require('./getRawData');
```
Comment (weizmm) [inline src/util/data_synchronizeData.js]: git pull

File: src/util/diffsHandlerUtils/addedDataHandler.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+const {sendLog, logLevel} = require('../logger');
+const logDetails = require('../logDetails');
+const domainUserHandler = require('../domainUserHandler');
+const Auth = require('../../auth/auth');
+
+require('dotenv').config();
+
+/*
+ * diffsObj - object that contain the results of diffs checking (added,updated,same,removed & all)
+ * dataSourceperson_ready_for_kartoffel - string the express the name of the data source
```
Comment (weizmm) [inline src/util/diffsHandlerUtils/addedDataHandler.js]: fix me

File: src/schedule.js [hunk 8]
```diff
@@ -8,19 +8,51 @@ const Redis = require("ioredis");
...
-    app.use(xls)
-    app.listen(5000, () => console.log(`Example app listening on port 5000!`))
+    app.use(xls);
+    app.listen(5000, () => console.log(`Example app listening on port 5000!`));
 }
 
-const scheduleTime = process.env.NODE_ENV === 'production' ? fn.runningTime : new Date().setMilliseconds(new Date().getMilliseconds() + 200);
+schedule.scheduleJob(scheduleRecoveryTime, async () => {
+    let akaData = await getRawData(fn.dataSources.aka);
+    akaRecovery(akaData);
```
Comment (weizmm) [inline src/schedule.js]: add comment that explained why aka needed spacial flow

File: src/schedule.js [hunk 9]
```diff
@@ -8,19 +8,51 @@ const Redis = require("ioredis");
...
+
+    const esData = await getRawData(fn.dataSources.es);
+    const adsData = await getRawData(fn.dataSources.ads);
+    const adnnData = await getRawData(fn.dataSources.adNN);
+    const lmnData = await getRawData(fn.dataSources.lmn);
+    const mdnData = await getRawData(fn.dataSources.mdn);
+    const mmData = await getRawData(fn.dataSources.mm);
+
+    const adjustEsData = await Promise.all(esData.map(async person => await matchToKartoffel(person, fn.dataSources.es)));
```
Comment (weizmm) [inline src/schedule.js]: change to promiseAllWithFails

File: src/util/recoveryUtils/compareToKartoffel.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+const added = require('../diffsHandlerUtils/addedDataHandler');
+const currentUnit_to_DataSource = require('../createDataSourcesMap');
+
```
Comment (weizmm) [inline src/util/recoveryUtils/compareToKartoffel.js]: add comment /**

File: src/util/updateSpecificFields.js [hunk 11]
```diff
@@ -4,10 +4,11 @@ const {sendLog, logLevel} = require('./logger');
 const logDetails = require('../util/logDetails');
 const fn = require('../config/fieldNames');
 const Auth = require('../auth/auth');
+const _ = require('lodash');
```
Comment (weizmm) [inline src/util/updateSpecificFields.js]: remove

File: src/util/getRawData.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
+const p = require('../config/paths');
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/util/getRawData.js]: done

File: src/util/data_synchronizeData.js [hunk 13]
```diff
@@ -1,18 +1,33 @@
...
+    // solve the problem that if runnig the module twice at same time on the clock
+    if (files[files.length - 1] === `${actionDescription}_${dateAndTime}.log` || files[files.length - 1] === 'archive') {
+        const completeFiles = fs.readdirSync(`${path}/archive/`);
+        lastJsonName = (completeFiles[completeFiles.length - 1]);
+    }
+
 
     // save the new json as file in the server and get the name of the kast file
-    let previous_data_file_name = saveAsFile(data, `./data/${dataSource}`, `${dataSource}_raw_data`);
+    // saveAsFile(data, path, actionDescription);
```
Comment (weizmm) [inline src/util/data_synchronizeData.js]: delete the comments

File: src/util/diffsHandler.js [hunk 14]
```diff
@@ -10,6 +10,7 @@ require('dotenv').config();
  */
 
 module.exports = async (diffsObj, dataSource, aka_all_data) => {
+    // promise all with fails
```
Comment (weizmm) [inline src/util/diffsHandler.js]: change to promise all

File: src/util/saveAsFile.js [hunk 15]
```diff
@@ -19,32 +18,32 @@ const pathHandler = (path) => {
...
         })
     }
     catch (err) {
-        sendLog(logLevel.error, logDetails.error.ERR_SAVE_DATA_FILE, actionDescription, dateAndTime, err.message);        
+        sendLog(logLevel.error, logDetails.error.ERR_SAVE_DATA_FILE, actionDescription, dateAndTime, err.message);
         return err.message;
     };
+    // let lastJsonName = files[files.length - 1]
+    // // solve the problem that if runnig the module twice at same time on the clock
+    // if (files[files.length - 1] === `${actionDescription}_${dateAndTime}.log` || files[files.length - 1] === 'archive') {
```
Comment (weizmm) [inline src/util/saveAsFile.js]: delete comments

**Pull request merged**